Zend/zend_execute.c

```diff
@@ -1795,6 +1795,18 @@ static zend_never_inline void zend_fetch_dimension_address_UNSET(zval *result, z
 	zend_fetch_dimension_address(result, container_ptr, dim, dim_type, BP_VAR_UNSET);
 }
 
+static zend_always_inline int zend_fetch_dimension_address_read_valid_container(zval *container, int support_strings) {
+	if (EG(current_execute_data)->opline->opcode != ZEND_FETCH_LIST) {
+		if (EG(current_execute_data)->opline->op1_type != IS_VAR) {
```

Comment:
Our standard approach to suppressing duplicate warnings is use of ERROR zvals. Doing this will need some care to ensure there is no impact on the fast-path though.